allennlp/data/dataset_readers/coreference_resolution/winobias.py

```diff
@@ -0,0 +1,167 @@
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import Field, ListField, TextField, SpanField, MetadataField, SequenceLabelField
+from allennlp.data.instance import Instance
+from allennlp.data.tokenizers import Token
+from allennlp.data.token_indexers import SingleIdTokenIndexer, TokenIndexer
+from allennlp.data.dataset_readers.dataset_utils import  enumerate_spans
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DatasetReader.register("winobias")
+class WinobiasReader(DatasetReader):
+    """
+    TODO(Mark): Add paper reference.
+
+    Winobias is a dataset to analyse the issue of gender bias in co-reference
+    resolution. It contains simple sentences with pro/anti stereotypical gender
+    associations with which to measure the bias of a coreference system trained
+    on another corpora.
+
+    Returns a list of ``Instances`` which have four fields: ``text``, a ``TextField``
+    containing the full sentence text, ``spans``, a ``ListField[SpanField]`` of inclusive start and
+    end indices for span candidates, and ``metadata``, a ``MetadataField`` that stores the instance's
+    original text. For data with gold cluster labels, we also include the original ``clusters``
+    (a list of list of index pairs) and a ``SequenceLabelField`` of cluster ids for every span
+    candidate.
+
+    Parameters
+    ----------
+    max_span_width: ``int``, required.
+        The maximum width of candidate spans to consider.
+    token_indexers : ``Dict[str, TokenIndexer]``, optional
+        This is used to index the words in the sentence.  See :class:`TokenIndexer`.
+        Default is ``{"tokens": SingleIdTokenIndexer()}``.
+    """
+    def __init__(self,
+                 max_span_width: int,
+                 token_indexers: Dict[str, TokenIndexer] = None,
+                 lazy: bool = False) -> None:
+        super().__init__(lazy)
+        self._max_span_width = max_span_width
+        self._token_indexers = token_indexers or {"tokens": SingleIdTokenIndexer()}
+
+    @overrides
+    def _read(self, file_path: str):
+        # if `file_path` is a URL, redirect to the cache
+        file_path = cached_path(file_path)
+
+        for sentence in open(file_path, "r"):
+            tokens = sentence.strip().split(" ")
+            clusters: DefaultDict[int, List[Tuple[int, int]]] = collections.defaultdict(list)
+            words = []
+            inside_square_span = False
+            inside_round_span = False
+            for index, token in enumerate(tokens):
+                # Coreference is annotated using [square brackets]
+                # or (round brackets) around coreferent phrases.
+                if "[" in token and "]" in token:
+                    clusters[0].append((index, index))
+                elif "[" in token:
+                    clusters[0].append((index, index))
+                    inside_square_span = True
```

Comment:
Why do you need this flag?  Can't you just use the index when you get to `]`?  Also, looks like this logic won't work for nested annotations.  I assume you don't have those either?